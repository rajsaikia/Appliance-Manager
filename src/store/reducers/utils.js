import { cloneDeep } from 'lodash';

export const filterRooms = rooms => {
	return rooms.filter(res => res.name);
};

export const updateAppliances = (appliances, source) => {
	const newUpdateAppliances = cloneDeep(appliances);
	const orderAt = newUpdateAppliances.findIndex(o => o.id === source.id);
	const app = newUpdateAppliances.filter(res => res.id === source.id);
	const keynames = Object.keys(source);
	app[0].nickname = source.nickname;
	app[0].properties.map(res =>
		keynames.includes(res.name) ? (res.value = source[res.name]) : null
	);
	newUpdateAppliances[orderAt] = app[0];
	return newUpdateAppliances;
};

export const addNewAppliances = (appliances, payload) => {
	const newUpdateAppliances = cloneDeep(appliances);
	const data = {
		...payload,
		...{
			type: 'New Appliances',
			nickname: 'NA',
			properties: [
				{
					name: 'power',
					type: 'toggle',
					value: true,
				},
			],
		},
	};
	newUpdateAppliances.push(data);
	return newUpdateAppliances;
};

export const addAppliancesToRoom = (rooms, selectedRoom, payload) => {
	const newUpdateRooms = cloneDeep(rooms);
	newUpdateRooms
		.filter(res => res.id === selectedRoom)[0]
		.applianceIds.push(payload.id);
	return newUpdateRooms;
};
export const AddAppliancesIndex = (appliancesIndex, payload) => {
	const newUpdateRooms = cloneDeep(appliancesIndex);
	newUpdateRooms.push(payload.id);
	return newUpdateRooms;
};
