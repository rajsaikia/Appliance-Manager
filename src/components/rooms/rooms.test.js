import React from 'react';
import { render } from 'jest/utils';
import '@testing-library/jest-dom/extend-expect';
import Rooms from './index';

describe('<Checkbox /> - Component', () => {
	let container = null;
	const data = [
		{
			id: 1,
			name: 'Living Room',
			applianceIds: [1],
		},
		{
			id: 2,
			name: 'Kitchen',
			applianceIds: [2, 3, 4],
		},
		{
			id: 3,
			name: 'Laundry',
			applianceIds: [5, 6],
		},
		{
			id: 4,
			name: 'Home Office',
			applianceIds: [6, 7],
		},
	];

	const Component = () => {
		return <Rooms rooms={data} />;
	};
	beforeEach(async () => {
		container = render(<Component />);
	});

	afterEach(() => {
		container.unmount();
	});
	it('renders correctly', async () => {
		expect(container.asFragment()).toMatchSnapshot();
	});
	it('it should contain the text Living Room', async () => {
		expect(container.getByText('Living Room')).toBeInTheDocument();
	});
	it('it should contain the text Kitchen', async () => {
		expect(container.getByText('Kitchen')).toBeInTheDocument();
	});
	it('it should contain the text Laundry', async () => {
		expect(container.getByText('Laundry')).toBeInTheDocument();
	});
	it('it should contain the text Home Office', async () => {
		expect(container.getByText('Home Office')).toBeInTheDocument();
	});
});
