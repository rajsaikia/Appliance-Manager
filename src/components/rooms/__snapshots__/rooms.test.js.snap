// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> - Component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="row"
  >
    <div
      class="d-grid gap-2 d-md-flex justify-content-md-start roomSection"
    >
      <button
        class="btn btn-outline-secondary"
        type="button"
        value="0"
      >
        Living Room
      </button>
      <button
        class="btn btn-outline-secondary"
        type="button"
        value="1"
      >
        Kitchen
      </button>
      <button
        class="btn btn-outline-secondary"
        type="button"
        value="2"
      >
        Laundry
      </button>
      <button
        class="btn btn-outline-secondary"
        type="button"
        value="3"
      >
        Home Office
      </button>
    </div>
  </div>
</DocumentFragment>
`;
